dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +123,225 @@ async def get_cluster_status(self, req):
+
+        if state != "RUNNING":
+            raise ValueError(
+                f"The task attempt is not running: the current state is {state}."
+            )
+        logger.info(f"state {type(state)}: {state}")
+
+        return pid, ip
+
+    @routes.get("/task/traceback")
+    async def get_task_traceback(self, req) -> aiohttp.web.Response:
+        """
+        Retrieves the traceback information for a specific task.
+        We don't use pid and ip to get task cpu profile since ip and pid is bounded to a worker
+        and a worker may run different tasks at different time.
+        Therefore, we may provide wrong profile info since we don't know which task is running on the worker
+
+        Args:
+            req (aiohttp.web.Request): The HTTP request object.
+
+        Returns:
+            aiohttp.web.Response: The HTTP response containing the traceback information.
+
+        Raises:
+            ValueError: If the "task_id" parameter is missing in the request query.
+            ValueError: If the "attempt_number" parameter is missing in the request query.
+            ValueError: If the worker begins working on another task during the traceback retrieval.
+            aiohttp.web.HTTPInternalServerError: If there is an internal server error during the traceback retrieval.
+        """
+
+        if "task_id" not in req.query:
+            raise ValueError("task_id is required")
+        if "attempt_number" not in req.query:
+            raise ValueError("task's attempt number is required")
+
+        task_id = req.query.get("task_id")
+        attempt_number = req.query.get("attempt_number")
+
+        self.set_up_state_api()
+
+        try:
+            pid, ip = await self.get_worker_details_for_task(task_id, attempt_number)
+        except ValueError as e:
+            raise aiohttp.web.HTTPInternalServerError(text=str(e))
+
+        # Default not using `--native` for profiling
+        native = req.query.get("native", False) == "1"
+
+        logger.info(
+            "Sending stack trace request to {}:{} with native={}".format(
+                req.query.get("ip"), pid, native
+            )
+        )
+        reporter_stub = self._stubs[ip]
+        reply = await reporter_stub.GetTraceback(
+            reporter_pb2.GetTracebackRequest(pid=pid, native=native)
+        )
+
+        ## Get the new pid and ip again to check if the task is still running and the worker is still working on the task
+        ## Since there is a task scheduling strategy that a worker may run different tasks at different time
+        try:
+            new_pid, new_ip = await self.get_worker_details_for_task(
+                task_id, attempt_number
+            )
+        except ValueError as e:
+            raise aiohttp.web.HTTPInternalServerError(text=str(e))
+
+        if pid != new_pid or ip != new_ip:
+            raise ValueError(
+                f"The worker begin to work on other task, you could not get the correct CPU profile info"
```

Comment:
This might not be entirely true right? With current implementation, it could also be that the task simply finished/errored.

Reply:
@rickyyx I agreed, there are two edge case:
1. A task is not running
2. pid != new_pid or ip != new_ip, represent a worker begin to work on other tasks


And I catch edge case 1 in https://github.com/ray-project/ray/blob/2feecec9e536e90420cfb3c691455da14e39bea4/dashboard/modules/reporter/reporter_head.py#L177
 
Which is executed before `pid != new_pid or ip != new_ip`